joeynmt/prediction.py

```diff
@@ -382,76 +388,39 @@ def test(cfg_file,
             logger.info("Translations saved to: %s", output_path_set)
 
 
-def translate(cfg_file: str,
-              ckpt: str,
-              output_path: str = None,
-              batch_class: Batch = Batch,
-              n_best: int = 1) -> None:
+def load_params_for_prediction(cfg_file,ckpt: str):
+
     """
     Interactive translation function.
-    Loads model from checkpoint and translates either the stdin input or
-    asks for input to translate interactively.
-    The input has to be pre-processed according to the data that the model
-    was trained on, i.e. tokenized or split into subwords.
-    Translations are printed to stdout.
+    Loads the parameters and model needed to perform translation
 
-    :param cfg_file: path to configuration file
-    :param ckpt: path to checkpoint to load
-    :param output_path: path to output file
-    :param batch_class: class type of batch
-    :param n_best: amount of candidates to display
+    :param cfg_file: path to configuration file/ configuration file dictionary to load parameters
+    :param ckpt: path to checkpoint to load model
+   
     """
-
-    def _load_line_as_data(line):
-        """ Create a dataset from one line via a temporary file. """
-        # write src input to temporary file
-        tmp_name = "tmp"
-        tmp_suffix = ".src"
-        tmp_filename = tmp_name+tmp_suffix
-        with open(tmp_filename, "w", encoding="utf-8") as tmp_file:
-            tmp_file.write("{}\n".format(line))
-
-        test_data = MonoDataset(path=tmp_name, ext=tmp_suffix,
-                                field=src_field)
-
-        # remove temporary file
-        if os.path.exists(tmp_filename):
-            os.remove(tmp_filename)
-
-        return test_data
-
-    def _translate_data(test_data):
-        """ Translates given dataset, using parameters from outer scope. """
-        # pylint: disable=unused-variable
-        score, loss, ppl, sources, sources_raw, references, hypotheses, \
-        hypotheses_raw, attention_scores = validate_on_data(
-            model, data=test_data, batch_size=batch_size,
-            batch_class=batch_class, batch_type=batch_type, level=level,
-            max_output_length=max_output_length, eval_metric="",
-            use_cuda=use_cuda, compute_loss=False, beam_size=beam_size,
-            beam_alpha=beam_alpha, postprocess=postprocess,
-            bpe_type=bpe_type, sacrebleu=sacrebleu, n_gpu=n_gpu, n_best=n_best)
-        return hypotheses
-
-    cfg = load_config(cfg_file)
-    model_dir = cfg["training"]["model_dir"]
-
-    _ = make_logger(model_dir, mode="translate")
-    # version string returned
-
-    # when checkpoint is not specified, take oldest from model dir
-    if ckpt is None:
-        ckpt = get_latest_checkpoint(model_dir)
+    
+    params_dict={}
+    if isinstance(cfg_file,str):
```

Comment:
This block is duplicated - maybe it's worth factoring it out into a separate function in `helpers.py`.